rocksdb_admin/admin_handler.cpp

```diff
@@ -929,6 +974,177 @@ void AdminHandler::async_tm_addS3SstFilesToDB(
   callback->result(AddS3SstFilesToDBResponse());
 }
 
+int AdminHandler::getPartitionIDFromDBName(const std::string& db_name) {
+  const size_t n = db_name.find_first_of("0123456789");
+  if (n != std::string::npos){
+    return std::stoi(db_name.substr(n, std::string::npos));
+  }
+  LOG(ERROR) << "No partition id found in db name " << db_name;
+
+  return -1;
+}
+
+std::string AdminHandler::getHostName() {
```

Comment:
can we use ip instead? if yes, we can reuse function getLocalIPAddress() in network_util.h.